lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -14,13 +14,13 @@ def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
     end
 
     def sync
-      out, _err, st = kubectl.run("get", type, @name, "-a", "--output=json")
+      raw_json, _err, st = kubectl.run("get", type, @name, "-a", "--output=json")
       if @found = st.success?
-        pod_data = JSON.parse(out)
+        pod_data = JSON.parse(raw_json)
         interpret_json_data(pod_data)
       else # reset
         @status = @phase = nil
-        @ready = false
+        @ready = @found = false
```

Comment:
It's used by `exists?`... but I'm not sure why I moved it here if that's what you mean. It should already be set, so I'll change it back.